scripts/Jenkinsfile-deploy

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/groovy
+
+/*
+ * This file is part of the Software Factory project
+ * Copyright (C) Pelagicore AB
+ * SPDX-License_identifier: LGPL-2.1
+ * This file is subject to the terms of the LGPL-2.1 license.
+ * Please see the LICENSE file for details.
+ */
+
+//
+// Please note, this will only work on the Jenkins set up used for pelux.io
+// The only reasons for it being in this repo are 1) it has to do with the PELUX
+// Software Factory, and 2) it keeps the script under version control rather
+// than on the Jenkins instance.
+//
+// In short, this script looks for all tags on the remote, and builds the SWF
+// for them, plus master. If there has been changes since last successful build,
+// we build all tags and branches, and deploy them to the web server
+//
+
+// We run this on any node that has Sphinx and web server access
+node('Sphinx && WWW') {
+
+    // For scope reasons we define it here
+    def buildList = null
+
+    // We copy everything to a staging directory to avoid 404 while the job is running
+    String stagingDirectory = "/var/www/software-factory-staging/"
+    String swfDirectory = '/var/www/software-factory/'
+
+    // A function for finding all tags and branches on the remote
+    // We also save the result of this scan to a file which we archive between
+    // builds to be able to compare refs between builds
+    String targetFinder = '''
+        TAGS=$(git ls-remote --tags --refs origin | tee -a builds.txt | awk '{ print $2 }')
+        for TAG in $TAGS; do
+            echo ${TAG##refs/tags/}
+        done
```

Comment:
Would like to see this written in groovy instead if possible.

Reply:
I'll see what I can do.